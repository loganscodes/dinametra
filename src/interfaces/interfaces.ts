export interface Welcome {
    collection: Collection;
}

export interface Collection {
    version:  string;
    href:     string;
    items:    Item[];
    
}

export interface Item {
    href:   string;
    data:   Datum[];
    links?: ItemLink[];
}

export interface Datum {
    title:              string;
    keywords:           string[];
    nasa_id:            string;
    date_created:       Date;
    description:        string;
    description_508?:   string;
    secondary_creator?: string;
    location?:          string;
    album?:             string[];
    photographer?:      string;
}



export interface ItemLink {
    href:    string;
}


