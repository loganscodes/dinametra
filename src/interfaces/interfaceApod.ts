export interface ApodData {
    date: string;
    explanation: string;
    hdurl: string;
    media_type: string;
    service_version: string;
    title: string;
    url: string;
    copyright: string;
}

export interface ApodProps {
    apod: {
        title: string;
        url: string;
        copyright: string;
        date: string;
        explanation: string;
    };
}