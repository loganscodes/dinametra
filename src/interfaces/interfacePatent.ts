export interface PatentData {
    id: string;
    title: string;
    patent: string;
    abstract: string;
    contact: string;
    date: number;
}