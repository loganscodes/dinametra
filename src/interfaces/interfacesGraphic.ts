import { ChangeEvent } from "react";

export interface GrapProps {
    data: any
}


export interface RangePickerProps {
    startDateI: string,
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
    endDateI: string
    onChangeEnd: (event: ChangeEvent<HTMLInputElement>) => void;
}