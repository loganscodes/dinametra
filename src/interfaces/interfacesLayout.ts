import { MouseEventHandler, ReactNode } from "react";

export interface MenuProps{
    onClick: MouseEventHandler<HTMLButtonElement>
}

export interface ItemNavProps{
    path: string;
    name: string
}

export interface LayoutProps {
    children: ReactNode
}