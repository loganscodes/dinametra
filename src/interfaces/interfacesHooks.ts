import { Item } from "./interfaces";

export interface PagProps{
    notices: Item[];
    filterByYear: (notice: Item) => boolean 
}

export interface VideoNasaProps {
    selectedNoticeHref: string;
}