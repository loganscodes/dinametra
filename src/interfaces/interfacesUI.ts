import { ChangeEvent, ChangeEventHandler, MouseEventHandler } from "react"
import { Item } from "./interfaces"

export interface UIButtonProps{
    onClick?: MouseEventHandler<HTMLButtonElement>
    text?: string
    tooltipID?: string
    tooltipContent?: string
    type?: "submit" | "reset" | "button"
    className?: string
}

export interface UIImgProps{
    tooltipID: string,
    tooltipContent: string,
    src: string
}

export interface UIInputProps {
    value: string
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
    tooltipID: string,
    tooltipContent: string
    placeholder: string
}

export interface UIInputDateProps{
    title: string
    value: string,
    onChange: (event: ChangeEvent<HTMLInputElement>) => void;
}

export interface UIPaginationProps {
    onClickPrevPage: MouseEventHandler<HTMLButtonElement>;
    onClickNextPage: MouseEventHandler<HTMLButtonElement>;
    currentPage: number;
    totalPages: number;
    
}

export interface UIParProps {
    content: string | number
    tooltipID?: string
    tooltipContent?: string
    className: string
    useDangerousHTML?: boolean  
}

export interface UISearchProps{
    selectedYear: string
    notices: Item[]
    onChange: ChangeEventHandler<HTMLSelectElement>;
}

export interface UITitleProps {
    tag: "h1" | "h2" | "h3" | "h4" | "h5" ;
    title: string;
    tooltipID: string
    tooltipContent: string;
    className: string
}

export interface UITitlePatentProps{
    tooltipContent: string
    content: string
    tooltipID: string 
}

export interface UIVideoProps {
    source: string
}
