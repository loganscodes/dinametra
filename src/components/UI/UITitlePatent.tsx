import { Tooltip } from 'react-tooltip'
import { UITitlePatentProps } from '../../interfaces/interfacesUI'



const UITitlePatent = ({tooltipContent, content, tooltipID}:UITitlePatentProps) => {
    return (
        <>
            <h3 tabIndex={0}
                data-tooltip-id={tooltipID}
                data-tooltip-content={tooltipContent}
                data-tooltip-place="top"
                className='text-3xl font-bold'
                dangerouslySetInnerHTML={{ __html: content }}
                aria-label={content} />
            
            <Tooltip id={tooltipID} />
        </>

    )
}

export default UITitlePatent