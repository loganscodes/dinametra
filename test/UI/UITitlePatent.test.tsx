import React from 'react';
import { UITitlePatentProps } from '../../src/interfaces/interfacesUI'
import UITitlePatent from '../../src/components/UI/UITitlePatent'
import renderer from 'react-test-renderer';


describe('UITitlePatent', () => {
    test('renders correctly with props', () => {
        const props: UITitlePatentProps = {
            tooltipContent: 'Tooltip Content',
            content: 'Title Content',
            tooltipID: 'title-tooltip',
        };

        const component = renderer.create(<UITitlePatent {...props} />);
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
});