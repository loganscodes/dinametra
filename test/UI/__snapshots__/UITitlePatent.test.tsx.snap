// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UITitlePatent renders correctly with props 1`] = `
<h3
  aria-label="Title Content"
  className="text-3xl font-bold"
  dangerouslySetInnerHTML={
    {
      "__html": "Title Content",
    }
  }
  data-tooltip-content="Tooltip Content"
  data-tooltip-id="title-tooltip"
  data-tooltip-place="top"
  tabIndex={0}
/>
`;
